resources/migrations/001_update_migrations.yaml

```diff
@@ -9279,26 +9279,27 @@ databaseChangeLog:
 # 1) Think through some of these points when writing a migration, learn from our past mistakes:
 #    - Do you really need a migration? Could the feature work without it? Prefer not to if possible.
 #      Data in the wild can be vastly different from what you expect, and it's hard to get right.
-#    - If you still need to, make sure it works when Metabase is encrypted. Fields like `metabase_database.details`
-#      or `setting.value` can be encrypted, so you need to decrypt them in your migration. See #42617, #44048.
-#    - Prefer SQL migrations over custom migrations.
-#    - Still, make sure your migration is backward compatible: it might not be possible to add a constraint back
-#      if you are dropping it in a migration.
+#    - Make sure your migration is backward compatible: it might not be possible to add a constraint back
+#      if you drop it in a migration.
 #    - Postgres, MySQL and H2 have their differences. Make sure your migration works for all.
-#    - Custom migrations and their tests demand a high level of reliability. They should only use code from the standard library.
-#      Requiring code from the Metabase codebase is a bad idea because it can change and break your migration.
-#      More in `metabase.db.custom_migrations` namespace doc.
-#    - Custom migrations shouldn't use Toucan models; work directly with the table names.
+#    - Database encryption is a major issue:
+#      - Fields like `metabase_database.details` or `setting.value` can be encrypted, so you need to decrypt them in
+#        your migration. See #42617, #44048.
+#      - Database could be partially encrypted, see https://www.notion.so/72575933ef2a446bafd16909e05a7387
+#    - Custom migrations:
+#      - Prefer SQL migrations when possible.
+#      - Never use application code: it can change and *will* break your migration.
+#      - Do not use Toucan models - refer table names directly.
+#      - More in `metabase.db.custom_migrations` namespace doc.
+#    - Never delete a migration: users won't be able to downgrade. If you really need to, see #44908
+#    - Do not backport `v51....` to version 50:
+#      - Metabase will try to downgrade it.
+#      - Instead, write a migration with a target version in mind
```

Comment:
How about something like this?

```suggestion
#    - Migration ID must match its earliest released version.
#      - If the migration will not be backported, use the next version.
#      - Otherwise, use the oldest version consistently every PR.
#      - This prevents the migration being rolled back unintentionally when downgrading.
```

What do we do if we decide to backport to version X-1 after we've already released it to version X though?

Reply:
> What do we do if we decide to backport to version X-1 after we've already released it to version X though?

cry in the corner first I guess, no other good options here :)